src/jaxsim/api/link.py

```diff
@@ -31,15 +32,16 @@ def name_to_idx(model: js.model.JaxSimModel, *, link_name: str) -> jtp.Int:
         The index of the link.
     """
 
-    if link_name in model.kin_dyn_parameters.link_names:
-        return (
-            jnp.array(
-                np.argwhere(np.array(model.kin_dyn_parameters.link_names) == link_name)
-            )
-            .squeeze()
-            .astype(int)
+    if link_name not in model.link_names():
+        raise ValueError(f"Link '{link_name}' not found in the model.")
+
+    return (
+        jnp.array(
+            np.argwhere(np.array(model.kin_dyn_parameters.link_names) == link_name)
         )
-    return jnp.array(-1).astype(int)
+        .astype(int)
+        .squeeze()
+    )
```

Comment:
Same as above. Be aware that `numpy` is no more used
```suggestion
    return jnp.array(model.kin_dyn_parameters.link_names.index(link_name)).squeeze()
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/182/commits/039f97fd8ca4095badc1bf57598d925d4c937043